### DISCUSSION TITLE
Is subresource filter affected by the manifest v3 limit?

### DISCUSSION THREAD

**Discussion by wchen342** [bromite/bromite#1234] [Q&A]
I was trying to figure this out but cannot find information anywhere. I know for extensions the limits are hard-coded somewhere related to `MAX_GETMATCHEDRULES_CALLS_PER_INTERVAL` and some other constants but I am not sure whether they are used in subresource filters too.

Btw I search around and see the v3 change landed in v88 but I couldn't find what uBlock Origin did to adapt the change. The issue about Manifest [doesn't have any updates](https://github.com/uBlockOrigin/uBlock-issues/issues/338). Do you know anything about it?

@csagan5 @uazo

**Comment by csagan5** [comment]
Are you working on some subresource filter engine modification for ungoogled-chromium?

> I am not sure whether they are used in subresource filters too.

It would need some major research in the code, but I do not think they are used.

> I couldn't find what uBlock Origin did to adapt the change. The issue about Manifest [doesn't have any updates](https://github.com/uBlockOrigin/uBlock-issues/issues/338). Do you know anything about it?

They use manifest v2 and when v2 is not supported anymore it will stop working.

**Reply by wchen342** [reply]
> Are you working on some subresource filter engine modification for ungoogled-chromium?

No, I was just thinking of how much value there is to have a desktop fusion of ungoogled-chromium and bromite. If uBlock stops working in the future there will be need for a native adblocker on chromium.
About that, are you keeping Bromite on Android because you want to focus your development? Or is there any other reason?

**Reply by csagan5** [reply]
> how much value there is to have a desktop fusion of ungoogled-chromium and bromite. 

I think that UC has most of the Bromite patches, so there is no real need for a Bromite for desktop.

> If uBlock stops working in the future there will be need for a native adblocker on chromium.

Bromite has some Android UI to use the subresource filter, but the engine is untouched.

> wow, that would be a big problem.
> it will probably be necessary to act on the policies, who knows

Not a problem for Bromite, there are no extensions.

**Reply by uazo** [reply]
> I think that UC has most of the Bromite patches

you say? at first glance it doesn't seem like it, but I should check it carefully

> there is no real need for a Bromite for desktop

i tried uc in the desktop some time ago, long before you know bromite, i was scared by the fact that despite all the blocks i have in my pc, suddenly a malicious extension was installed without my knowledge.
so I switched to chromium unpatched, in the (mistaken) belief that extensions and flags made me safe from privacy issues, but seeing the patches we are doing I realized that some things cannot be turned off by any extension or flag.
so, let's talk about a desktop version of bromite (even without extensions), because I have no intention of installing brave.

**Reply by wchen342** [reply]
Exactly the reason I started the discussion. I am curious though about the malicious extension you get. It shouldn't be possible to install extensions on it at all except for manually install from `chrome://extensions` or using `chromium-web-store`, so I am not sure how you get the extension. That is indeed strange. If somehow an extension managed to install itself without user action then Bromite won't save you too, that is more like a security problem.

**Reply by uazo** [reply]
>  I am curious though about the malicious extension you get. 

I don't remember the circumstances exactly.

> It shouldn't be possible to install extensions on it at all except for manually install from chrome://extensions or using chromium-web-store,

actually the security of extensions is something I'd like to delve into with the code.
I'm not very sure about that you say, because I think a change on the configuration files is enough to install an extension, after all those files are not protected, and cr somewhere has to write that information. in android it is so, the files are editable, only the sandbox of the s.o. protects them.

from what I knew UC does not have access to the store, or am I wrong?
I think that in chromium, at least, extension must be signed by the store which, although it is not a guarantee of security, is a first obstacle to overcome, but, I don't know how they dealt with that problem, for now, I trust them.

**Reply by csagan5** [reply]
> > I think that UC has most of the Bromite patches
> 
> you say? at first glance it doesn't seem like it, but I should check it carefully

It is true that Bromite has more patches but I have myself no idea about how it compares to the Chromium forks that preceded it and are still around these days.

> from what I knew UC does not have access to the store, or am I wrong?

It does download extensions from the store.



> I think that in chromium, at least, extension must be signed by the store which, although it is not a guarantee of security, is a first obstacle to overcome, but, I don't know how they dealt with that problem, for now, I trust them.

Some "malware extensions" are legit on the store, for example those that "enrich" your search results. Those could be installed on your browser even without your consent.
Also, don't know if this article was debunked or not, but software on desktop can read your cookies because there is no sandbox: https://web.archive.org/web/20200828044302/https://www.threatspike.com/blog/zoom_cookies.html

**Comment by uazo** [comment]
it seems that everything passes by `APIPermissionID::kWebRequestBlocking`, here a comment on it

https://source.chromium.org/chromium/chromium/src/+/main:extensions/browser/api/web_request/web_request_api.cc;l=1072

with some work you could go back to the enterprise policy mentioned in the documentation and understand how it really works.

the subresource filter code does not seem to be affected.
@csagan5 remember the edge adblock talk? now maybe i know where to attach it... "did you read the license?" no :)

> They use manifest v2 and when v2 is not supported anymore it will stop working.

wow, that would be a big problem.
it will probably be necessary to act on the policies, who knows

> how much value there is to have a desktop fusion of ungoogled-chromium and bromite

what a next big step

> Or is there any other reason?

Time for me!

**Reply by wchen342** [reply]
> > how much value there is to have a desktop fusion of ungoogled-chromium and bromite
> what a next big step

It is actually quite simple, nothing comparable to extensions. Bromite adblocker are basically patches and since uc already has desktop versions, all that needed is just applying those additional patches.

I wonder how long enterprise policies will last. Have they ever got removed?

**Reply by uazo** [reply]
> It is actually quite simple

but do not forget all the java code to be reported in the native equivalent? maybe I'm missing something ...

**Reply by wchen342** [reply]
Ah right. But I had a quick look at the patches and it seems most of the adblock stuff are in native code, just need to add some UI stuff I think.

**Reply by wchen342** [reply]
The problem with my previous extension approach is I tried to build everything on top of existing chromium code, which leads to rebase hell each update. A separate module detached to chromium base code will be much better. I wonder how much work is needed to update `Bromite-AdBlockUpdaterService.patch` each version?

**Reply by csagan5** [reply]
> I wonder how long enterprise policies will last. Have they ever got removed?

The enterprise policies do not matter IMO: after v2 is disabled the relative code will be deleted, so it will be no more a policy problem.

> I wonder how much work is needed to update `Bromite-AdBlockUpdaterService.patch` each version?

Almost nothing, it is one of the least maintenance-intensive patches. But desktop users are a different crowd than mobile users: on desktop you would need a much better UI and customization options, most users will ask you to compete with uBlock feature set (be aware of what you embark on).
